portafly/src/components/data-list/utils/filterRows.ts

```diff
@@ -0,0 +1,29 @@
+import { DataListRow, DataListCol } from 'types'
+
+export function filterRows(
+  rows: DataListRow[],
+  filters: Record<string, string[]>,
+  columns: DataListCol[]
+) {
+  let newRows = [...rows]
+
+  Object.keys(filters).forEach((categoryName) => {
+    if (newRows.length === 0 || filters[categoryName].length === 0) {
+      return
+    }
+
+    const colIdx = columns.findIndex((c) => c.categoryName === categoryName)
+
+    if (colIdx === -1) {
+      console.warn('Category "%s" doesn\'t correspond to any column in your DataList', categoryName)
+      return
+    }
+
+    newRows = newRows.filter((r) => {
+      const value = (r.cells)[colIdx].toLowerCase()
+      return filters[categoryName].some((f) => value.indexOf(f.toLowerCase()) > -1)
+    })
+  })
+
+  return newRows
+}
```

Comment:
This could be more declarative instead of imperative as I've shown in the previous review ;)

Reply:
Yeah.. but I really don't like chaining so many arrow functions. It makes it harder to debug and read IMO.
I think this way it's more readable at least, even if it's a bit verbose. You think there's something wrong with it or is it only preference?

Reply:
Besides the verbosity, the use of `let`, `forEach` and 2 conditionals? just the code being imperative. The functions being named help for the readability, the name is telling you what's doing, and/or what it is. Using that kind of composition makes it easier to debug and maintain, each tiny function is easier than 20+ lines one. 
That being said and this being tested, right now it's ok to merge... Most probably it will be follow by a refactor more declarative in the near future